sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -1133,27 +1119,30 @@ private Mono<PagedResponseBase<MessagesDequeueHeaders, QueueMessageItem>> transf
         if (queueMessageInternalItems == null) {
             queueMessageInternalItems = Collections.emptyList();
         }
+
+        // creating queue client to pass in when the catch block is executed
+        QueueClient queueClient = new QueueClient(client, queueName, accountName, serviceVersion, messageEncoding,
```

Comment:
We should not create a new client each time this method is called. Create one instance of sync client and use it here.